nilearn/datasets/tests/test_utils.py

```diff
@@ -316,6 +316,22 @@ def test_uncompress():
             shutil.rmtree(dtemp)
 
 
+def test__safe_extract():
+    # Test vulnerability patch by mimicking path traversal
+    dtemp = mkdtemp()
+    ztemp = os.path.join(dtemp, 'test.tar')
+    fd, temp = mkstemp(dir=dtemp)
```

Comment:
it might be clearer to give it a name and writes somehting in it like `in_archive_file = tmp_path / "something.txt"; in_archive_file.write_text("hello")`